astropy/coordinates/baseframe.py

```diff
@@ -1798,8 +1801,11 @@ def separation(self, other):
         from .angles import Angle, angular_separation
 
         self_unit_sph = self.represent_as(r.UnitSphericalRepresentation)
-        other_transformed = other.transform_to(self)
-        other_unit_sph = other_transformed.represent_as(r.UnitSphericalRepresentation)
+        other_unit_sph = (
+            getattr(other, "frame", other)
```

Comment:
It may still be worthwhile to define a simple `_to_own_frame` so that these pieces of code are kept together. E.g., possibly one should just do `other.frame if isinstance(other, SkyCoord) else other` or `getattr(other, "_sky_coord_frame", other)` [as is done in `transform_to`]. But, really, also happy to just have it like this.

Reply:
OK, let's keep it as is here!